api/api-materials-v1/src/main/java/com/thoughtworks/go/apiv1/materials/MaterialsAdminControllerV1.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.materials;
+
+import com.thoughtworks.go.api.ApiController;
+import com.thoughtworks.go.api.ApiVersion;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
+import com.thoughtworks.go.api.util.GsonTransformer;
+import com.thoughtworks.go.config.exceptions.BadRequestException;
+import com.thoughtworks.go.config.exceptions.HttpException;
+import com.thoughtworks.go.server.materials.MaterialUpdateService;
+import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult;
+import com.thoughtworks.go.spark.Routes;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static spark.Spark.*;
+
+@Component
+public class MaterialsAdminControllerV1 extends ApiController implements SparkSpringController {
```

Comment:
Maybe a different controller name?

Reply:
Does `MaterialsNotifyControllerV1` sound better? The base path is `/api/admin/materials`, so there's a possibility that endpoints other than `/notify` may end up in this controller.

Reply:
`MaterialsNotifyControllerV1` sounds better. Agreed that there is a possibility that the controller can have other endpoints, I think whenever we add new endpoint at that time we can change it to something else. - WDYT?